units/src/amount/signed.rs

```diff
@@ -65,6 +65,17 @@ impl SignedAmount {
     /// Constructs a new [`SignedAmount`] with satoshi precision and the given number of satoshis.
     pub const fn from_sat(satoshi: i64) -> SignedAmount { SignedAmount(satoshi) }
 
+    /// Constructs a new [`SignedAmount`] with satoshi precision and the given number of satoshis.
+    ///
+    /// Warning, it's possible to violate the [`SignedAmount`] range invariant.  That is, there is
+    /// no check that [`SignedAmount`] is greater than [`SignedAmount::MIN`] nor that the value it
+    /// represents is less than [`SignedAmount::MAX`].  If you wish to represent values that are
+    /// not in range, consider using a different type.  Note that this function may be modified in
+    /// the future to disallow violations of the [`SignedAmount`] range invariant.
```

Comment:
In 73e3a8fbf26e9852f2b70bea864757f662d12b07:

This is overly wordy and also incorrect (it suggests that the MIN..MAX range is exclusive rather than inclusive). Better would be "If the passed value is lower than SignedAmount::MIN or greater than SignedAmount::MAX your code will misbehave in unspecified ways. If you wish to represent values outside this range you should directly use a numeric type".

Reply:
I mean, someone may be using this to represent values for a different ledger like we had discussed in a different thread.  In that case, they should use a type specific to their ledger instead of a numeric type.  Maybe I'm splitting hairs here but numeric type may not be what they should use instead.  What about this:

```
    /// Warning, it's possible to violate the [`SignedAmount`] range invariant.  If the value
    /// passed is lower than [`SignedAmount::MIN`] or greater than [`SignedAmount::MAX`], your code
    /// will misbehave in unspecified ways.  If you wish to represent values outside this range,
    /// you should use a different type.  Note that this function may be modified in the future to
    /// disallow violations of the [`SignedAmount`] range invariant.
```

Reply:
I don't like this "may be modified to disallow violations". This is confusingly specific. Is the implication that we won't make other changes that affect behavior when the user passes an out-of-range value?

Reply:
Would it be better to just be super generic and say: `Note that this function may be modified in the future` ?  I think the only possible modification is to enforce the range invariant but we could not be that specific I guess.

Reply:
maybe we say, beware, this function could be changed in unpredictable ways :P

Reply:
No, we don't need to say anything about this specific function. We already say that if the user violates the invariant then their code will malfunction. We do not need to say how. We do not need to promise it will be consistent. We do not need to imply that these malfunctions may change in particular ways.

Reply:
oh ok.  My original copy didn't include any such language however there was some comments above about adding "We reserve the right to..".  Maybe I misunderstood the previous feedback.  I'll drop the last line.